File: backend/src/main/java/dev/minimtr/model/dto/TransitConfig.java
package dev.minimtr.model.dto;

import java.util.List;
import java.util.Map;

public record TransitConfig(String name, String timezone, String adapter, String mode,
        Mapping mapping, Simulation simulation, Realtime realtime) {
    public record Mapping(String routesDataset, String stationsDataset,
            Map<String,String> routeFields, Map<String,String> stationFields, List<Pattern> patterns,
            double snapToleranceMeters) {}
    public record Pattern(String code, String featureKey, String direction, boolean reversed, List<Stop> stops) {}
    public record Stop(String stationKey, Double fraction) {}
    public record Simulation(String serviceDayStart, List<Rule> rules, List<TimetableTrip> timetable) {}
    public record Rule(String id, String pattern, String first, String last, double headwaySeconds,
            double speedKmph, double dwellSeconds) {}
    public record Realtime(List<Monitor> monitors, int staleAfterSeconds) {}
    public record Monitor(String line, String station) {}
}

File: front/src/api/transitConfig.ts
export interface Pattern {
    code: string; featureKey: string; direction: string; reversed: boolean;
    stops: {stationKey: string; fraction: number | null}[];
}
export interface TimetableTrip {
    key: string; pattern: string; serviceDate: string | null;
    stops: {seq:number; arrival:string|null; departure:string|null}[];
}
export interface TransitConfig {
    name:string; timezone:string; adapter:string; mode:'simulation'|'realtime';
    mapping: { routesDataset:string; stationsDataset:string; routeFields:Record<string,string>; stationFields:Record<string,string>;
        patterns:Pattern[]; snapToleranceMeters:number };
    simulation: {serviceDayStart:string; rules:{id:string;pattern:string;first:string;last:string;headwaySeconds:number;speedKmph:number;dwellSeconds:number}[]; timetable:TimetableTrip[]};
    realtime: {monitors:{line:string;station:string}[];staleAfterSeconds:number};
}
export interface Configured {code:string; id:number; config:TransitConfig}
